### DISCUSSION TITLE
ES8388 Headphones Volume too Low - Modification possible?

### DISCUSSION THREAD

**Discussion by baesek** [marcel-licence/esp32_midi_sampler#81] [Q&A]
Hi, 
I tried around alot now to get a louder signal out of the headphones.

At first I thought that the solution is too set the registers right as I found this comment in the Ai-Thinker-Open repository: https://github.com/Ai-Thinker-Open/ESP32-A1S-AudioKit/issues/26#issuecomment-1219363637
But I tried it with almost all settings now, so I start to believe, that the ES8388 Chip is just a shit, and the headphone output is just for testing....
My last hope is that it might be possible to just wire the output of the NS4150 http://inder.cn/WJXZ/nsiway/InderNS4150EN.PDF
amp used for each speaker to the headphone jack to get a decent volume. 
Do you think that's possible? I don't want to damage anything, and my knowledge about the electronics is not good enough to say it will work. Couldn't it just be done by attenuating the amplifiers output via two resistors per channel? or with little trimpots?

This is the schematic of the speaker amplifier:
[esp32-audio-kit_v2.2_sch.pdf](https://github.com/marcel-licence/esp32_midi_sampler/files/9459708/esp32-audio-kit_v2.2_sch.pdf)

Here is a schematic which shows a connection of headphones in parallel to speaker output:
https://www.ti.com/lit/ds/symlink/tpa6030a4.pdf?ts=1661932852771&ref_url=https%253A%252F%252Fwww.google.com%252F
-paul

**Comment by marcel-licence** [comment]
Hi,

I also prefer the Audio Kit with the AC101. I tried just headphones and external amplifier yet. I am using the headphones output as a line out. If you use speaker you should use the two white connectors. If you use an external amplifier you should try to go from the 3.5mm headphones output. 
In addition to that the volume could be also a bit low caused by the software. I tried to keep some headroom for higher polyphony. Adding multiple signals can lead very quickly into distortion. 
I can imagine that you might get some ideas from other projects.

Best regards,
Marcel

**Reply by baesek** [reply]
Hi,
I am still trying to make a progress using the esp32-a1s (ES8388 or AC101) and I am thinking about buying some standalone A1S boards with AC101. My question is, is the headphone output which comes directly from the ESP32-A1S Board with the AC101 Audio Codec Board louder than the ES8388?
best,
paul

**Reply by marcel-licence** [reply]
There is an amplifier placed on the audio kit. I am not sure if that could/should be used for the headphones. The white connectors are used for speakers. The amplifier can be switched on via a specific GPIO. Does this help?
Unfortunately I do not have an comparison at this moment. I would recommend to generate a testsignal to ensure that you have the loudest signal possible.

**Reply by baesek** [reply]
I just got an LyraT board which also uses the ES8388 Chip and to my surprise the audio output is much louder and stronger on this thing. Not super loud, but loud enough to feel the energy of a bassdrum. The only problem is that it is a little distorted in high gain settings. This problem is described here: https://www.esp32.com/viewtopic.php?t=9375

Now I am wondering how the hell this is possible that the same chip with the same code gives so different results. At first I thought the reason was that I always powered the A1S ES8388 Board through its UART Micro USB Port instead of the Power Port but it made no difference, when I tested it. 
Perhaps the difference is in the schematics / circuit around the ES8388 chip.
LyraT (source: https://dl.espressif.com/dl/schematics/esp32-lyrat-v4.3-schematic.pdf)
<img width="1472" alt="Bildschirmfoto 2022-09-22 um 12 53 24" src="https://user-images.githubusercontent.com/10615191/191728639-d19a458e-119c-41df-afa8-9cfedd03d986.png">


A1S ES8388 (source: https://docs.ai-thinker.com/_media/esp32-a1s_v2.3_specification.pdf)
<img width="1201" alt="Bildschirmfoto 2022-09-22 um 12 52 14" src="https://user-images.githubusercontent.com/10615191/191903539-91597e36-0ce3-4b53-b4cf-afdc02937385.png">


The first significant difference is that on the A1S Board the LOUT1, LOUT2, ROUT1 and ROUT2 are passed through a 22µF Capacitor and a 33 Ohm Resistor whereas on the Lyrat Board they are just passed through one 1µF Capacitor. 

The second difference I saw in the first gaze is that the LyraT Board has a 10µF + a 0.1µF buffer capacitor for the DVDD Pin whereas the A1S has only two 0.1µF capacitors there.

This is from the ES8388 Datasheet under Typical Application:
<img width="729" alt="Bildschirmfoto 2022-09-22 um 14 00 06" src="https://user-images.githubusercontent.com/10615191/191741236-c278354c-27a8-4aed-ae09-be0d0c7f051e.png">
